huaweicloud/services/cfw/resource_huaweicloud_cfw_alarm.go

```diff
@@ -0,0 +1,271 @@
+	alarm, err := GetAlarm(readAlarmClient, fwInstanceID, alarmType)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving CFW alarm")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("alarm_type", utils.PathSearch("alarm_type", alarm, nil)),
+		d.Set("alarm_time_period", utils.PathSearch("alarm_time_period", alarm, nil)),
+		d.Set("frequency_count", utils.PathSearch("frequency_count", alarm, nil)),
+		d.Set("frequency_time", utils.PathSearch("frequency_time", alarm, nil)),
+		d.Set("severity", utils.PathSearch("severity", alarm, nil)),
+		d.Set("topic_urn", utils.PathSearch("topic_urn", alarm, nil)),
+		d.Set("language", utils.PathSearch("language", alarm, nil)),
+		d.Set("username", utils.PathSearch("username", alarm, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func GetAlarm(readAlarmClient *golangsdk.ServiceClient, fwInstanceID string, alarmType int) (interface{}, error) {
+	readAlarmHttpUrl := "v1/{project_id}/cfw/alarm/config?fw_instance_id={fw_instance_id}"
+	readAlarmPath := readAlarmClient.Endpoint + readAlarmHttpUrl
+	readAlarmPath = strings.ReplaceAll(readAlarmPath, "{project_id}", readAlarmClient.ProjectID)
+	readAlarmPath = strings.ReplaceAll(readAlarmPath, "{fw_instance_id}", fwInstanceID)
+
+	readAlarmOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	resp, err := readAlarmClient.Request("GET", readAlarmPath, &readAlarmOpt)
+	if err != nil {
+		return nil, common.ConvertExpected400ErrInto404Err(err, "error_code", "CFW.00200005")
+	}
+
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	findAlarmStr := fmt.Sprintf("alarm_configs[?alarm_type==`%d`]|[0]", alarmType)
+	alarm := utils.PathSearch(findAlarmStr, respBody, nil)
+	status := int(utils.PathSearch("enable_status", alarm, float64(2)).(float64))
+	// If the status of the alarm is off or not found.
+	if status == 0 || status == 2 {
+		return nil, golangsdk.ErrDefault404{}
+	}
+
+	return alarm, nil
+}
+
+func resourceAlarmDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		deleteAlarmHttpUrl = "v1/{project_id}/cfw/alarm/config?fw_instance_id={fw_instance_id}"
+		deleteAlarmProduct = "cfw"
+	)
+	deleteAlarmClient, err := cfg.NewServiceClient(deleteAlarmProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CFW client: %s", err)
+	}
+
+	fwInstanceID := d.Get("fw_instance_id").(string)
+	deleteAlarmPath := deleteAlarmClient.Endpoint + deleteAlarmHttpUrl
+	deleteAlarmPath = strings.ReplaceAll(deleteAlarmPath, "{project_id}", deleteAlarmClient.ProjectID)
+	deleteAlarmPath = strings.ReplaceAll(deleteAlarmPath, "{fw_instance_id}", fwInstanceID)
+
+	alarmType := d.Get("alarm_type").(int)
+	_, err = GetAlarm(deleteAlarmClient, fwInstanceID, alarmType)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving CFW alarm")
+	}
```

Comment:
when delete the resource, terraform will call Read func first, so we don't need to check it here
```suggestion
```